### PR TITLE
Feature/helm addons

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:*

*Description of changes:*
- upgraded version of CDK to 1.132
- metrics server now is using helm
- aws lb controller has iam policy in-place
- appmesh addon, awslb and metrics server are using base class
- argocd takes values for bootstrap app
- nginx upgraded
- helm and manifest support and add-on implementation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/argocd/application.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import * as dot from 'dot-object';
+import { GitOpsApplicationDeployment, GitRepositoryReference } from '../../spi';
+
+export class ArgoApplication {
```
Comment (kcoleman731) [inline lib/addons/argocd/application.ts:4]: docs.

File: lib/addons/aws-loadbalancer-controller/iam-policy.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
+export const AwsLoadbalancerControllerIamPolicy = {
```
Comment (kcoleman731) [inline lib/addons/aws-loadbalancer-controller/iam-policy.ts:1]: awesome

File: lib/addons/helm-addon/kubectl-provider.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import { KubernetesManifest } from "@aws-cdk/aws-eks";
+import { Construct } from "@aws-cdk/core";
+import { ClusterInfo, Values } from "../..";
+
+
+export interface HelmChartConfiguration {
```
Comment (kcoleman731) [inline lib/addons/helm-addon/kubectl-provider.ts:6]: docs

File: lib/addons/helm-addon/index.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+export abstract class HelmAddOn implements spi.ClusterAddOn {
+
+    constructor(readonly props: HelmAddOnProps) {}
+    
+    abstract deploy(clusterInfo: spi.ClusterInfo): void | Promise<Construct>;
+
+    protected addHelmChart(clusterInfo: spi.ClusterInfo, values?: spi.Values ) : Construct {
+        const kubectlProvider = new KubectlProvider(clusterInfo);
+        const chart = {...this.props, ...{ values: values ?? {}} };
```
Comment (kcoleman731) [inline lib/addons/helm-addon/index.ts:16]: nit - consider moving the ternary out of map literal for ease of reading.

File: lib/addons/helm-addon/kubectl-provider.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+export interface ManifestDeployment extends Omit<ManifestConfiguration, "manifestUrl"> {
+    manifest: any,
+    values: Values
+}
+
+export class KubectlProvider {
+
+    constructor(private readonly clusterInfo : ClusterInfo) {}
+
+    public static helmProvider = function(clusterInfo: ClusterInfo, props: HelmChartDeployment) : Construct {
```
Comment (kcoleman731) [inline lib/addons/helm-addon/kubectl-provider.ts:61]: consider renaming function to match CDK function name. I.E. `addHelmChart`.

**Review by kcoleman731 — COMMENTED** [review]
Looking good!

**Review by askulkarni2 — COMMENTED** [review]
looking good..

File: lib/addons/argocd/application.ts (hunk 1 above)
Comment (shapirov103) [reply lib/addons/argocd/application.ts:4]: addressed

File: lib/addons/helm-addon/kubectl-provider.ts (hunk 3 above)
Comment (shapirov103) [reply lib/addons/helm-addon/kubectl-provider.ts:6]: Addressed

File: lib/addons/helm-addon/index.ts (hunk 4 above)
Comment (shapirov103) [reply lib/addons/helm-addon/index.ts]: Addressed

File: lib/addons/helm-addon/kubectl-provider.ts (hunk 5 above)
Comment (shapirov103) [reply lib/addons/helm-addon/kubectl-provider.ts]: addressed

File: lib/addons/appmesh/index.ts [hunk 6]
```diff
@@ -74,27 +83,23 @@ export class AppMeshAddOn implements ClusterAddOn {
...
+            tracing: {
+                enabled: this.options.enableTracing,
+                provider: this.options.tracingProvider,
+                address: this.options.tracingAddress,
+                port: this.options.tracingPort
             }
-        });
+        };
 
+        values = merge(values, this.props.values ?? {});
```
[the remark this answers is not in the document]
Comment (shapirov103) [reply lib/addons/appmesh/index.ts:100]: yes, good point. I have applied to Argo at the moment. For simple value merge we could still spread operator.

**Comment by shapirov103** [comment]
accidentally pruned

**Review by kcoleman731 — COMMENTED** [review]
LGTM lets run e2e.

File: lib/addons/appmesh/index.ts [hunk 7]
```diff
@@ -35,20 +37,27 @@ export interface AppMeshAddOnProps {
...
     readonly options: AppMeshAddOnProps;
 
     constructor(props?: AppMeshAddOnProps) {
-        this.options = { ...defaultProps, ...props };
+        super({ ...defaultProps, ...props });
+        this.options = this.props;
     }
 
-    deploy(clusterInfo: ClusterInfo): void {
+    override deploy(clusterInfo: ClusterInfo): void {
```
Comment (askulkarni2) [inline lib/addons/appmesh/index.ts:60]: > When a method is marked with override, TypeScript will always make sure that a method with the same name exists in a the base class.

Good to know!

**Review by askulkarni2 — APPROVED** [review]
Great PR!

**Pull request merged**